tests/test_streaming.py

```diff
@@ -59,6 +59,65 @@ def test_dataloader_epoch_size_no_streams(local_remote_dir: Tuple[str,
             assert samples_seen == epoch_size
 
 
+@pytest.mark.parametrize('batch_size', [4])
+@pytest.mark.parametrize('seed', [2222])
+@pytest.mark.parametrize('shuffle', [False, True])
+@pytest.mark.parametrize('drop_last', [False])
+@pytest.mark.parametrize('num_workers', [3, 6])
+@pytest.mark.parametrize('num_canonical_nodes', [4, 8])
+@pytest.mark.parametrize('epoch_size', [10, 200])
+@pytest.mark.usefixtures('local_remote_dir')
+def test_dataloader_fixed_sampling(local_remote_dir: Any, batch_size: int, seed: int,
+                                   shuffle: bool, drop_last: bool, num_workers: int,
+                                   num_canonical_nodes: int, epoch_size: int):
+    remote_dir, local_dir = local_remote_dir
+    convert_to_mds(out_root=remote_dir,
+                   dataset_name='sequencedataset',
+                   num_samples=117,
+                   size_limit=1 << 8)
+
+    # Build StreamingDataset
+    dataset = StreamingDataset(local=local_dir,
+                               remote=remote_dir,
+                               shuffle=shuffle,
+                               batch_size=batch_size,
+                               shuffle_seed=seed,
+                               num_canonical_nodes=num_canonical_nodes,
+                               epoch_size=epoch_size,
+                               sampling_method='fixed')
+
+    # Build DataLoader
+    dataloader = StreamingDataLoader(dataset=dataset,
+                                     batch_size=batch_size,
+                                     num_workers=num_workers,
+                                     drop_last=drop_last)
+
+    # iterate once over the dataloader (first epoch)
+    first_samples_seen = {}
```

Comment:
Can you combine below two for loops into one ?

Reply:
bad merge again...